pallets/messages/src/lib.rs

```diff
@@ -76,8 +82,14 @@ pub mod pallet {
 		/// Weight information for extrinsics in this pallet.
 		type WeightInfo: WeightInfo;
 
-		/// A type that will supply account related information.
-		type AccountProvider: AccountProvider<AccountId = Self::AccountId>;
+		/// A type that will supply MSA related information
+		type MsaInfoProvider: MsaLookup + MsaValidator<AccountId = Self::AccountId>;
+
+		/// A type that will supply delegation related information
+		type DelegationInfoProvider: ProviderLookup + DelegationValidator;
+
+		/// A type that will validate schema grants
+		type SchemaGrantValidator: SchemaGrantValidator;
```

Comment:
In theory we could combine all 5 into one config member:
```
`AccountProvider: MsaLookup + MsaValidator + ProviderLookup + DelegationValidator + SchemaGrantValidator
```

But I think it would be unwieldy.